File: src/com/peyo/launcherlb/AppInfo.java
package com.peyo.launcherlb;

import android.content.ComponentName;
import android.content.Context;
import android.content.pm.ActivityInfo;
import android.content.pm.PackageManager;
import android.graphics.drawable.Drawable;

public class AppInfo {

    private final PackageManager mPm;
    private final ActivityInfo mActivityInfo;
    private final ComponentName mComponentName;

    AppInfo(Context context, ActivityInfo info) {
        mPm = context.getPackageManager();
        mActivityInfo = info;
        mComponentName = new ComponentName(info.packageName, info.name);
    }

    public  CharSequence getLabel() {
        return mActivityInfo.loadLabel(mPm);
    }

    public Drawable getIcon() {
        return mActivityInfo.loadIcon(mPm);
    }

    public Drawable getBanner() {
        return mActivityInfo.loadBanner(mPm);
    }

    public ComponentName getComponentName() {
        return mComponentName;
    }

}

File: src/com/peyo/rplauncher/AppInfo.kt
package com.peyo.rplauncher

import android.content.ComponentName
import android.content.Context
import android.content.pm.ActivityInfo
import android.content.pm.PackageManager
import android.graphics.drawable.Drawable

class AppInfo internal constructor(context: Context, private val mActivityInfo: ActivityInfo) {

    private val mPm: PackageManager
    val componentName: ComponentName

    val label: CharSequence?
        get() = mActivityInfo.loadLabel(mPm)

    val icon: Drawable?
        get() = mActivityInfo.loadIcon(mPm)

    val banner: Drawable?
        get() = mActivityInfo.loadBanner(mPm)

    init {
        mPm = context.packageManager
        componentName = ComponentName(mActivityInfo.packageName, mActivityInfo.name)
    }

}
